labs/lab05/frontend/lib/core/auth/auth_service.dart

```
@@ -94,8 +94,39 @@ class AuthService {
   // - Return AuthResult.success on successful authentication
   // - Return AuthResult.networkError if any exception occurs during the process
   Future<AuthResult> login(String email, String password) async {
-    // TODO: Implement this method
-    throw UnimplementedError('AuthService login not implemented');
+    try {
+      final emailError = FormValidator.validateEmail(email);
+      final passwordError = FormValidator.validatePassword(password);
+
+      if (emailError != null || passwordError != null) {
+        return AuthResult.validationError;
+      }
+
+      final sanitizedEmail = FormValidator.sanitizeText(email);
+
+      final user = await _userRepository.findByEmail(sanitizedEmail);
+      if (user == null) {
+        return AuthResult.invalidCredentials;
+      }
+
+      final passwordValid =
+          await _userRepository.verifyPassword(sanitizedEmail, password);
+      if (!passwordValid) {
+        return AuthResult.invalidCredentials;
+      }
+
+      final token = _jwtService.generateToken(user.id.toString(), user.email);
+      _currentState = AuthState(
+        isAuthenticated: true,
+        currentUser: user,
+        token: token,
+        loginTime: DateTime.now(),
+      );
+
+      return AuthResult.success;
+    } catch (_) {
+      return AuthResult.networkError;
```

Consider logging the caught exception or returning more detailed error information for debugging/troubleshooting, instead of swallowing all errors as network errors.